.github/workflows/linting.yml

```diff
@@ -160,3 +160,19 @@ jobs:
           skip: ./src/control/vendor,./src/control/go.sum,./.git
           ignore_words_file: ci/codespell.ignores
           builtin: clear,rare,informal,names,en-GB_to_en-US
+
+  linting-summary:
+    name: Linting Summary
+    runs-on: ubuntu-22.04
+    needs: [isort, shell-check, log-check, ftest-tags, flake8-lint, doxygen, pylint, codespell]
+    if: always()
+    steps:
+      - name: Check if any job failed
+        run: |
+          if [[ "$(echo "${{ join(needs.*.result, ' ') }}" \
+                | tr -s ' ' '\n' | sort -u)" == "success" ]]; then
+            echo "All jobs succeeded"
+          else
+            echo "One or more jobs did not succeed"
+            exit 1
+          fi
```

Comment:
Unless you really want the https://github.com/daos-stack/daos/blob/4af82fbb698271d79242382d2eafe35918b1057c/.github/workflows/linting.yml#L174 line printed, you can probably simplify this whole workflow down with:

```suggestion
    if: ! success()
    steps:
      - name: Check if any job failed
        run: |
            echo "One or more jobs did not succeed"
            exit 1
```

Or if you do still want https://github.com/daos-stack/daos/blob/4af82fbb698271d79242382d2eafe35918b1057c/.github/workflows/linting.yml#L174 this simplifies the test a bit:

```suggestion
    if: always() && !cancelled() # never use just always() as it creates un-cancellable jobs
    steps:
      - name: Check if any job failed
        run: |
          if [[ -z "$(echo "${{ join(needs.*.result, '') }}" | sed -e 's/success//g')" ]]; then
            echo "All jobs succeeded"
          else
            echo "One or more jobs did not succeed"
            exit 1
          fi
```

As noted in the suggestion above, using just `always()` in a job `if:` creates an un-cancellable job.  Probably not a terribly bad thing for such a small job as above, but a good practice non-the-less.  This is quite a bit more important in long running jobs of course.

Reply:
This is tricky because the GHA simply don't support what we need here. The problem with `cancelled()` is that if some other job from the `needs` was cancelled, then this summary is skipped. And GitHub branch protections count a "skip" as a pass. So this summary needs to ALWAYS run. That said, I agree with your point. Maybe what I should do here is keep `always()` and wrap the job with a timeout so it doesn't run forever?

Reply:
About the printed lines, I don't care. Thanks for the simplification

Reply:
Hmm. Maybe I can play with these some more
https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions
> [failure](https://docs.github.com/en/actions/learn-github-actions/expressions#failure)
Returns true when any previous step of a job fails. If you have a chain of dependent jobs, failure() returns true if any ancestor job fails.

Taking into account your suggestion, maybe this does what I want?
```
    if: ! success()
    steps:
      - name: Check if any job failed
        run: exit 1
    if: success()
    steps:
      - name: Check if any job failed
        run: exit 0
```

I'll fiddle with this some more

Reply:
It seems `!cancelled()` works well here. But we can't simplify to `!success()` because then the summary is "skipped" and branch protections count a skip as a pass. I'm not sure if a "cancelled" counts as a pass or fail, but that problem already exists today

Reply:
On second thought... Maybe `!success()` could work here.
If `!success()` evaluates to `False`, it means all dependent jobs passed. So if the Summary is skipped and counted as a pass, that's okay.
If `!success()` evaluates to `True`, it means some dependent job failed. So the Summary will run and also fail.

But I do feel more secure with the Summary always running so we aren't relying on Skip==Pass